internal/packages/pacman.go

```diff
@@ -0,0 +1,117 @@
+package packages
+
+import (
+	"bufio"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"patchmon-agent/pkg/models"
+
+	"github.com/sirupsen/logrus"
+)
+
+var installedPackageRe = regexp.MustCompile(`^(\S+)\s+(\S+)$`)
+var checkUpdateRe = regexp.MustCompile(`^(\S+)\s+(\S+)\s+->\s+(\S+)$`)
+
+// PacmanManager handles pacman package information collection
+type PacmanManager struct {
+	logger *logrus.Logger
+}
+
+// NewPacmanManager creates a new Pacman package manager
+func NewPacmanManager(logger *logrus.Logger) *PacmanManager {
+	return &PacmanManager{
+		logger: logger,
+	}
+}
+
+// indirections for testability
+var (
+	lookPath   = exec.LookPath
+	runCommand = exec.Command
+)
+
+// GetPackages gets package information for pacman-based systems
+func (m *PacmanManager) GetPackages() ([]models.Package, error) {
+	// Get installed packages
+	installedCmd := runCommand("pacman", "-Q")
+	installedOutput, err := installedCmd.Output()
+	var installedPackages map[string]string
+	if err != nil {
+		m.logger.WithError(err).Warn("Failed to get installed packages")
+		installedPackages = make(map[string]string)
+	} else {
+		installedPackages = m.parseInstalledPackages(string(installedOutput))
+	}
+
+	upgradablePackages, err := m.getUpgradablePackages()
+	if err != nil {
+		return nil, err
+	}
+
+	// Merge and deduplicate packages
+	packages := CombinePackageData(installedPackages, upgradablePackages)
+	return packages, nil
+}
+
+// getUpgradablePackages runs checkupdates and returns parsed packages.
+func (m *PacmanManager) getUpgradablePackages() ([]models.Package, error) {
+	if _, err := lookPath("checkupdates"); err != nil {
+		m.logger.WithError(err).Warn("checkupdates not found (pacman-contrib not installed)")
+		return nil, err
+	}
+
+	upgradeCmd := runCommand("checkupdates")
+	upgradeOutput, err := upgradeCmd.Output()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Fails when no updates are present. Excerpt from checkupdates manpage:

```
ERRORS
       On exit, checkupdates will return one of the following error codes.

       0
           Normal exit condition.

       1
           Unknown cause of failure.

       2
           No updates are available.
```

Reply:
Addressed in https://github.com/PatchMon/PatchMon-agent/pull/34/commits/1b494b62225bcb8a33e66dab4df7d89047055c62